backend/de.metas.material/cockpit/src/main/java/de/metas/material/cockpit/availableforsales/AvailableForSalesQuery.java

```diff
@@ -33,20 +35,45 @@
  */
 
 @Value
-@Builder
 public class AvailableForSalesQuery
 {
-	@NonNull ProductId productId;
+	@NonNull
+	ProductId productId;
 
-	@Nullable WarehouseId warehouseId;
+	@Nullable
+	WarehouseId warehouseId;
 
 	@NonNull
 	AttributesKeyPattern storageAttributesKeyPattern;
 
+	@NonNull
+	OrgId orgId;
+
 	@NonNull
 	Instant dateOfInterest;
 
 	int shipmentDateLookAheadHours;
 
 	int salesOrderLookBehindHours;
+
+	@Builder
+	public AvailableForSalesQuery(
+			@NonNull final ProductId productId,
+			@Nullable final WarehouseId warehouseId,
+			@NonNull final AttributesKeyPattern storageAttributesKeyPattern,
+			@NonNull final OrgId orgId,
+			@NonNull final Instant dateOfInterest,
+			final int shipmentDateLookAheadHours,
+			final int salesOrderLookBehindHours)
+	{
+		Check.assumeGreaterThanZero(orgId.getRepoId(), "OrgId must be regular!");
```

Comment:
- you can actually use Check.errorUnless(orgId.isRegular(), ...) (if you want)
- the check is fine in that it makes clear we need a "real" org-id and not "*", but we can add more context to give a possible supporter a better chance to see which ShipmentSchedule etc this is all about. Suggestion:
```java
		Check.errorUnless(orgId.isRegular(), "AD_Org_Id={} must be regular! M_Product_ID={}, M_Warehouse_ID={}, AttributesKey={}, dateOfInterest=",
						  OrgId.toRepoId(orgId), ProductId.toRepoId(productId), WarehouseId.toRepoId(warehouseId), storageAttributesKeyPattern, dateOfInterest);
```